tests/integration/cairo_compiler/prepare_files_fixture.py

```diff
@@ -0,0 +1,61 @@
+from typing import Callable
+from pathlib import Path
+from enum import Enum
+
+from tests.integration._conftest import ProtostarFixture
+
+TEST_CONTRACTS_PATH = Path(__file__).parent / "contracts"
+
+
+class RequestedFiles(Enum):
+    input_enum_contract_cairo = 1
+    input_basic_starknet_contract_cairo = 2
+    input_basic_starknet_test_cairo = 3
+    input_roll_test_cairo = 4
+    output_sierra = 5
+    output_casm = 6
+
+
+class PrepareFilesFixture:
+    def __init__(self, protostar: ProtostarFixture):
+        self.protostar = protostar
+
+    def prepare_files(self, requested_files: list[RequestedFiles]):
+        files = {}
+        for file in requested_files:
+            input_path = ""
+            if file == RequestedFiles.input_enum_contract_cairo:
+                input_path = TEST_CONTRACTS_PATH / "enum_contract.cairo"
+            elif file == RequestedFiles.input_basic_starknet_contract_cairo:
+                input_path = TEST_CONTRACTS_PATH / "basic_starknet_contract.cairo"
+            elif file == RequestedFiles.input_basic_starknet_test_cairo:
+                input_path = TEST_CONTRACTS_PATH / "basic_starknet_test.cairo"
+            elif file == RequestedFiles.input_roll_test_cairo:
+                input_path = TEST_CONTRACTS_PATH / "roll_test.cairo"
+            contents = ""
+            if input_path:
+                contents = Path(input_path).read_text()
+            file_with_ext = ".".join(file.name.rsplit("_", 1))
+            files[file.name] = (Path(f"./src/{file_with_ext}"), contents)
+
+        self.protostar.create_files(
```

Comment:
I don't understand why we need whole protostar project for those tests, it seems redundant

Reply:
https://github.com/software-mansion/protostar/issues/1465